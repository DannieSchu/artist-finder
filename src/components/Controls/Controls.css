.Controls {
  input {
    border-radius: .5rem;
    padding-left: 1rem;
    width: 15rem;
    height: 2rem;
  }
}
