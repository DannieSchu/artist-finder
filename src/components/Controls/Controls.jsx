import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ onChange, onSubmit, value, placeholder, buttonText = 'Submit' }) => {
  return (
    <form onSubmit={onSubmit}>
      <input 
        type="text" 
        value={value} 
        onChange={onChange} 
        placeholder={placeholder} />
      <button>{buttonText}</button>
    </form>
  );
};

Controls.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  buttonText: PropTypes.string
};

export default Controls;
