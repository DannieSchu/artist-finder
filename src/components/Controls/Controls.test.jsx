import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls.jsx';

describe('Controls', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Controls
        onSubmit={() => {}}
        onChange={() => {}}
        value=""
        placeholder="Enter an artist"
        buttonText="Go!" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
