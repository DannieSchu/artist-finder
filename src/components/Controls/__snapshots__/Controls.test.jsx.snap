// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls matches a snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Enter an artist"
    type="text"
    value=""
  />
  <button>
    Go!
  </button>
</form>
`;
